src/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -244,16 +244,30 @@ TritonLoader::StartTriton()
   REPORT_TRITONSERVER_ERROR(
       api_version_fn_(&api_version_major, &api_version_minor),
       "unable to get api version");
+
+  auto createErrorMessage = [](const std::string& message, int expected,
+                               int actual) {
+    return message + "Expected version: " + std::to_string(expected) + "\n" +
+           "Actual version: " + std::to_string(actual) + "\n";
+  };
+
   if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
-      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
-    std::stringstream sstream;
-    sstream << "triton server API version mismatch. \n"
-            << "Expected version major:" << TRITONSERVER_API_VERSION_MAJOR
-            << ", minor:" << TRITONSERVER_API_VERSION_MINOR << "\n"
-            << "  Actual version major:" << api_version_major
-            << ", minor:" << api_version_minor;
-    return Error(sstream.str());
+      (api_version_minor > TRITONSERVER_API_VERSION_MINOR)) {
```

Comment:
You're printing an error message for a major version mismatch when there's only a minor version mismatch here, I think. Be careful with the branching. You want separate messages for the three cases. You can also be more specific with the error messages this way too (e.g. state that the server API version is newer than the client version for that case).

Reply:
Oh makes sense!